Kconfig.zephyr

```diff
@@ -344,6 +344,24 @@ config BUILD_OUTPUT_S19
 	  Build an S19 binary zephyr/zephyr.s19 in the build directory.
 	  The name of this file can be customized with CONFIG_KERNEL_BIN_NAME.
 
+config BUILD_OUTPUT_UF2
+	bool "Build a binary in UF2 format"
+	depends on BUILD_OUTPUT_BIN
+	help
+	  Build a UF2 binary zephyr/zephyr.uf2 in the build directory.
+	  The name of this file can be customized with CONFIG_KERNEL_BIN_NAME.
+
+if BUILD_OUTPUT_UF2
+
+config BUILD_OUTPUT_UF2_FAMILY_ID
+	string "UF2 Device Family"
```

Comment:
I suggest to create a default list in `modules/Kconfig.uf2` as shown below and then add a line in https://github.com/zephyrproject-rtos/zephyr/blob/master/modules/Kconfig
```
source "modules/Kconfig.uf2"
```

```
config BUILD_OUTPUT_UF2_FAMILY_ID
	# Prompt less config to setup a list of default UF2 Family IDs 
	default "0x68ed2b88" if SOC_SERIES_SAMD21
	default "0x55114460" if SOC_SERIES_SAMD51
	default "0xada52840" if SOC_NRF52840_QIAA
	default "0x4fb2d5bd" if SOC_SERIES_IMX_RT
	default "0x2abc77ec" if SOC_SERIES_LPC55XXX
	default "0x647824b6" if SOC_SERIES_STM32F0X
	default "0x5ee21072" if SOC_STM32F103XE
	default "0x5d1a0a2e" if SOC_SERIES_STM32F2X
	default "0x6b846188" if SOC_SERIES_STM32F3X
	default "0x57755a57" if SOC_STM32F401XC
	default "0x57755a57" if SOC_STM32F401XE
	default "0x6d0922fa" if SOC_STM32F407XE
	default "0x8fb060fe" if SOC_STM32F407XG
	default "0x53b80f00" if SOC_SERIES_STM32F7X
	default "0x300f5633" if SOC_SERIES_STM32G0X
	default "0x4c71240a" if SOC_SERIES_STM32G4X
	default "0x6db66082" if SOC_SERIES_STM32H7X
	default "0x202e3a91" if SOC_SERIES_STM32L0X
	default "0x1e1f432d" if SOC_SERIES_STM32L1X
	default "0x00ff6919" if SOC_SERIES_STM32L4X
	default "0x04240bdf" if SOC_SERIES_STM32L5X
	default "0x70d16653" if SOC_SERIES_STM32WBX
	default "0x1c5f21b0" if SOC_ESP32
	help
	  Default UF2 Family IDs for known SoCs.
	  All values on this list must correspond to the values found in  scripts/uf2conv.py.
```

This allows us to not pollute the `Kconfig.zephyr` but still have a common default list of known boards without updating each board file.

**Note**: I propose `modules/Kconfig.uf2` although this is strictly not module code.
But I feel `subsys` or `lib` are worse alternative, and as there is no actual config settings (only default values), I feel this is best place.
Better proposals are welcome.

Reply:
`soc/Kconfig{,.uf2}` if we want them all in one place.

Does:

    default "0x57755a57" if SOC_STM32F401XC || SOC_STM32F401XE

work?  And sort them, either by value or by `SOC_` selector.

Reply:
> Does:
> 
> ```
> default "0x57755a57" if SOC_STM32F401XC || SOC_STM32F401XE
> ```
> 
> work? And sort them, either by value or by `SOC_` selector.

It sure does, I didn't notice there were duplicated default values.
Sorting is a good idea, so I updated proposal above accordingly, and I think based on SOC makes most sense, as the default value is supposed to be a random number.

Will update proposal accordingly.

> `soc/Kconfig{,.uf2}` if we want them all in one place.

Thanks for the proposal. This is also a possibility.
I think i'm split between this and the `modules`.
Small recap:
- `modules/Kconfig.uf2` clearly indicates that this is unrelated to core Zephyr SoC in general, like other module code, cons is that this is not pure module code.
- `soc/Kconfig.soc` indicates that the default settings for UF2 Family ID is related to SoC names, but i'm unsure if this will be first place people will look if looking for default `BUILD_OUTPUT_UF2_FAMILY_ID` values.